### DISCUSSION TITLE
Router with VPN client connected to VPN server. But PC under the router can't go online through VPN connection.

### DISCUSSION THREAD

**Discussion by hunhun010164** [strongswan/strongswan#822] [Q&A]
**Hardware:** TPLink TL-MR11U, changed to 32M ROM & 64M RAM
**Firmware:** Because this HW is not supported from V18. So the latest official version 17.01.7, r4030-6028f00df0 is used.
Since the ROM is still quite small, only very basic packages were installed: openssl-util strongswan-minimal strongswan-mod-curve25519 strongswan-mod-af-alg strongswan-mod-openssl strongswan-mod-pem strongswan-mod-eap-identity strongswan-mod-eap-mschapv2
BTW, even I use full strongswan packages on another router, same problem.

Since this HW has only one ethernet port, wireless as both wireless client and access point. Ethernet port as LAN
**/etc/config/wireless:**
config wifi-device 'radio0'
	option type 'mac80211'
	option hwmode '11g'
	option path 'platform/ar933x_wmac'
	option htmode 'HT20'
	option disabled '0'
	option channel '6'
	option country 'US'

config wifi-iface
	option network 'wan'
	option ssid 'test'
	option device 'radio0'
	option mode 'sta'
	option key '12345678'
	option encryption 'psk2+ccmp'

config wifi-iface 'default_radio0'
	option device 'radio0'
	option network 'lan'
	option mode 'ap'
	option ssid 'AP'
	option encryption 'psk2+ccmp'
	option key '12345678'

**/etc/config/network:**
config interface 'loopback'
	option ifname 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option ula_prefix 'fd8b:0835:56e3::/48'

config interface 'lan'
	option type 'bridge'
	option ifname 'eth0'
	option proto 'static'
	option ipaddr '192.168.100.1'
	option netmask '255.255.255.0'
	list dns '8.8.8.8'
	list dns '8.8.4.4'
	option ip6assign '60'

config interface 'wan'
	option proto 'dhcp'

**Main part of /etc/config/firewall:**
config defaults
	option syn_flood '1'
	option input 'ACCEPT'
	option output 'ACCEPT'
	option forward 'ACCEPT'

config zone
	option name 'lan'
	option input 'ACCEPT'
	option output 'ACCEPT'
	option forward 'ACCEPT'
	option network 'lan'

config zone
	option name 'wan'
	option input 'ACCEPT'
	option output 'ACCEPT'
	option forward 'ACCEPT'
	option masq '0'
	option mtu_fix '1'
	option network 'wan'

config forwarding
	option src 'lan'
	option dest 'wan'

config rule
	option name 'Dropbear'
	option src 'wan'
	option proto 'tcp'
	option dest_port '22'
	option target 'ACCEPT'

**ipsec.conf:**
config setup
	uniqueids = no

conn SH
	keyexchange=ikev2
	right=my.domain
	rightsubnet=0.0.0.0/0
	leftsourceip=%config
	leftsendcert=never
	leftauth=eap-mschapv2
	eap_identity=test
	auto=route

**Now the situation is:**
Router can connect VPN server, can resolve the domain, can ping internet address, traffic goes through VPN instead of WAN after checking by "traceroute". Good.

**The problem is:**
Only when "ipsec stop", the PC under router(no matter connected to LAN port or wireless) can get a valid IP (192.168.100.X) and go online.
Once "ipsec start", the PC under router(no matter connected to LAN port or wireless) can still keep the IP (192.168.100.X), but can't go online, and traffice can't go through VPN.

**What I want:**
After the router connected to VPN, PCs under the router can go online and all traffic go through router's VPN connection. Thanks.

**Comment by tobiasbrunner** [comment]
Please have a look at [ForwardingAndSplitTunneling](https://wiki.strongswan.org/projects/strongswan/wiki/ForwardingAndSplitTunneling).
